src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1351,9 +1402,60 @@ private void getVars(CORINFO_METHOD_STRUCT_* ftn, ref uint cVars, ILVarInfo** va
             // Just tell the JIT to extend everything.
             extendOthers = true;
         }
+
         private void setVars(CORINFO_METHOD_STRUCT_* ftn, uint cVars, NativeVarInfo* vars)
         {
-            // TODO: Debugging
+            if (_localSlotToInfoMap == null && _parameterIndexToNameMap == null)
```

Comment:
Do we ever have case where no parameter while having locals only?
Did you validate if the current logic works regardless of static (no this parameter) and instance method?